vpr/src/route/router_lookahead_map.cpp

```diff
@@ -424,135 +428,66 @@ static void compute_router_wire_lookahead(const std::vector<t_segment_inf>& segm
+        routing_cost_map.fill(Expansion_Cost_Entry());
 
-                int cost_index = rr_graph.node_cost_index(rr_node);
-                VTR_ASSERT(cost_index != OPEN);
+        for (auto& point : region.points) {
+            for (auto node_ind : point.nodes) {
+                //reset cost for this segment
+                RRNodeId sample_node(node_ind);
 
-                int seg_index = device_ctx.rr_indexed_data[cost_index].seg_index;
+                int sample_x = rr_graph.node_xlow(sample_node);
+                int sample_y = rr_graph.node_ylow(sample_node);
 
-                if (seg_index == iseg) {
-                    sample_nodes[chan_type].push_back(RRNodeId(inode));
+                if (rr_graph.node_direction(sample_node) == DEC_DIRECTION) {
+                    sample_x = rr_graph.node_xhigh(sample_node);
+                    sample_y = rr_graph.node_yhigh(sample_node);
                 }
 
-                if (sample_nodes[chan_type].size() >= ref_increments.size()) {
-                    break;
-                }
+                run_dijkstra(sample_node,
+                             sample_x,
+                             sample_y,
+                             routing_cost_map,
+                             &dijkstra_data);
             }
         }
 
-        //Finally, now that we have a list of sample locations, run a Djikstra flood from
-        //each sample location to profile the routing network from this type
-
-        t_dijkstra_data dijkstra_data;
-        t_routing_cost_map routing_cost_map({device_ctx.grid.width(), device_ctx.grid.height()});
-
-        for (e_rr_type chan_type : {CHANX, CHANY}) {
-            if (sample_nodes[chan_type].empty()) {
-                VTR_LOG_WARN("Unable to find any sample location for segment %s type '%s' (length %d)\n",
-                             rr_node_typename[chan_type],
-                             segment_inf[iseg].name.c_str(),
-                             segment_inf[iseg].length);
-            } else {
-                //reset cost for this segment
-                routing_cost_map.fill(Expansion_Cost_Entry());
-
-                for (RRNodeId sample_node : sample_nodes[chan_type]) {
-                    int sample_x = rr_graph.node_xlow(sample_node);
-                    int sample_y = rr_graph.node_ylow(sample_node);
-
-                    if (rr_graph.node_direction(sample_node) == DEC_DIRECTION) {
-                        sample_x = rr_graph.node_xhigh(sample_node);
-                        sample_y = rr_graph.node_yhigh(sample_node);
-                    }
-
-                    run_dijkstra(sample_node,
-                                 sample_x,
-                                 sample_y,
-                                 routing_cost_map,
-                                 &dijkstra_data);
-                }
+#if defined(VPR_USE_TBB)
+        all_costs_mutex.lock();
```

Comment:
Rather than manually locking/unlocking the mutex which is error prone. We should probably be using a [lock guard](https://www.threadingbuildingblocks.org/docs/doxygen/a00136.html) to ensure it is automatically unlocked in all scenarios.